In [2]:
import helper


HF_returns = helper.load_data('HW_Hedge_Fund.xlsx',1)
M_ER_country = helper.load_data('HW_World.xlsx')
Emp_factors = helper.load_data('HW_Factors.xlsx')
HP_DJ_Sto = helper.load_data('HW_DJIA_Prices.xlsx',4)

print(HF_returns.keys())
print(M_ER_country.keys())
print(Emp_factors.keys())     
print(HP_DJ_Sto.keys())






Index(['Date', 'HFRI 400 (US) Fund Weighted Composite Index (HFRI4FWC)',
       'HFRI 400 (US) EH: Long/Short Index (HFRI4ELS)',
       'HFRI 400 (US) EH: Equity Market Neutral Index (HFRI4EMN)',
       'HFRI 400 (US) EH: Fundamental Value Index (HFRI4EHV)',
       'HFRI 400 (US) Event-Driven Index (HFRI4ED)',
       'HFRI 400 (US) Macro Index (HFRI4M)',
       'HFRI 400 (US) Relative Value Index (HFRI4RV)'],
      dtype='object')
Index(['Date', 'Austria', 'Australia', 'Belgium', 'Canada', 'Denmark',
       'Finland', 'France', 'Germany', 'HongKong', 'Ireland', 'Italy', 'Japan',
       'Netherlands', 'New Zealand', 'Norway', 'Singapore', 'Spain', 'Sweden',
       'Switzerland', 'UK'],
      dtype='object')
Index(['Unnamed: 0', 'Mkt-RF', 'SMB', 'HML', 'Mom   ', 'RF'], dtype='object')
Index(['Date', 'MMM', 'AXP', 'AMGN', 'AAPL', 'BA', 'CAT', 'CVX', 'CSCO', 'KO',
       'DIS', 'DD', 'GS', 'HON', 'HD', 'IBM', 'INTC', 'JNJ', 'JPM', 'MCD',
       'MRK', 'MSFT', 'NKE', 'PG', 'CRM', 'BAC', 'UN

In [3]:
# HP_DJ_Sto['Date'] = pd.to_datetime(HP_DJ_Sto['Date'])
# HP_DJ_Sto['Name']
Emp_factors.rename(columns={Emp_factors.keys()[0]: 'Date'}, inplace=True)

# Emp_factors.head()#dataframe
# HF_returns.head()
# M_ER_country.head()#dataframe
# HP_DJ_Sto.head()
# Emp_factors['Date']
print(Emp_factors['Date'].dtype)#
print(M_ER_country['Date'].dtype)#
print(HF_returns['Date'].dtype)
print(HP_DJ_Sto['Date'].dtype)  

# 



int64
int64
datetime64[ns]
datetime64[ns]


In [28]:
print(Emp_factors['Date'])#factor
print(HF_returns['Date'])#hedge fund
print(M_ER_country['Date'])#world
print(HP_DJ_Sto['Date'])#DJIA



0      1970-01-01 00:00:00.000192701
1      1970-01-01 00:00:00.000192702
2      1970-01-01 00:00:00.000192703
3      1970-01-01 00:00:00.000192704
4      1970-01-01 00:00:00.000192705
                    ...             
1152   1970-01-01 00:00:00.000202301
1153   1970-01-01 00:00:00.000202302
1154   1970-01-01 00:00:00.000202303
1155   1970-01-01 00:00:00.000202304
1156   1970-01-01 00:00:00.000202305
Name: Date, Length: 1157, dtype: datetime64[ns]
0     2005-01-01
1     2005-02-01
2     2005-03-01
3     2005-04-01
4     2005-05-01
         ...    
216   2023-01-01
217   2023-02-01
218   2023-03-01
219   2023-04-01
220   2023-05-01
Name: Date, Length: 221, dtype: datetime64[ns]
0     1970-01-01 00:00:00.000199101
1     1970-01-01 00:00:00.000199102
2     1970-01-01 00:00:00.000199103
3     1970-01-01 00:00:00.000199104
4     1970-01-01 00:00:00.000199105
                   ...             
379   1970-01-01 00:00:00.000202208
380   1970-01-01 00:00:00.000202209
381   1970-01-01 00:00:

In [35]:
#Convert 'Date' column to datetime format for Emp_factors and M_ER_country
'''Didn't do yet, could be improved in the future'''
# import pandas as pd
# Emp_factors['Date'] = pd.to_datetime(Emp_factors['Date'])
# M_ER_country['Date'] = pd.to_datetime(M_ER_country['Date'])
# print(len(Emp_factors))
# print(len(M_ER_country))
# print(len(HF_returns))
# print(len(HP_DJ_Sto))

# # Verify the date formats
# print(Emp_factors['Date'])
# print(M_ER_country['Date'])

# # Display the first few rows of each dataframe to confirm the changes
# print("\nEmp_factors head:")
# print(Emp_factors[['Date']].head())
# print("\nM_ER_country head:")
# print(M_ER_country[['Date']].head())
# print("\nHF_returns head:")
# print(HF_returns[['Date']].head())
# print("\nHP_DJ_Sto head:")
# print(HP_DJ_Sto[['Date']].head())

"Didn't do yet, could be improved in the future"

1. CAPM

In [8]:
print(HF_returns.keys())
print(Emp_factors.keys())
print(M_ER_country.keys())
print(HP_DJ_Sto.keys())


Index(['Date', 'HFRI 400 (US) Fund Weighted Composite Index (HFRI4FWC)',
       'HFRI 400 (US) EH: Long/Short Index (HFRI4ELS)',
       'HFRI 400 (US) EH: Equity Market Neutral Index (HFRI4EMN)',
       'HFRI 400 (US) EH: Fundamental Value Index (HFRI4EHV)',
       'HFRI 400 (US) Event-Driven Index (HFRI4ED)',
       'HFRI 400 (US) Macro Index (HFRI4M)',
       'HFRI 400 (US) Relative Value Index (HFRI4RV)'],
      dtype='object')
Index(['Date', 'Mkt-RF', 'SMB', 'HML', 'Mom   ', 'RF'], dtype='object')
Index(['Date', 'Austria', 'Australia', 'Belgium', 'Canada', 'Denmark',
       'Finland', 'France', 'Germany', 'HongKong', 'Ireland', 'Italy', 'Japan',
       'Netherlands', 'New Zealand', 'Norway', 'Singapore', 'Spain', 'Sweden',
       'Switzerland', 'UK'],
      dtype='object')
Index(['Date', 'MMM', 'AXP', 'AMGN', 'AAPL', 'BA', 'CAT', 'CVX', 'CSCO', 'KO',
       'DIS', 'DD', 'GS', 'HON', 'HD', 'IBM', 'INTC', 'JNJ', 'JPM', 'MCD',
       'MRK', 'MSFT', 'NKE', 'PG', 'CRM', 'BAC', 'UNH', 'V

In [72]:
# ! pip install scipy
from scipy import stats
import pandas as pd
class hedgeFundClass:   
    def __init__(self, HF_returns, HF_name):
        self.HF_name = HF_name
        self.beta = 0
        self.alpha = 0
        self.t_stat = 0
        self.r_squared = 0
        self.HF_returns = HF_returns
    
    def __str__(self):
        return (
            f'HF_name: {self.HF_name}\n'
            f'beta: {self.beta}\n'
            f'alpha: {self.alpha}\n'
            f't_stat: {self.t_stat}\n'
            f'r_squared: {self.r_squared}\n'
        )

    def run_capm_regression(self, Market_premium):
        slope, intercept, r_value, p_value, std_err = stats.linregress(Market_premium, self.HF_returns)
        self.beta = slope
        self.alpha = intercept
        self.t_stat = intercept / std_err
        self.r_squared = r_value ** 2
        return {
            'HF_name': self.HF_name,
            'beta': slope,
            'alpha': intercept,
            't_stat': intercept / std_err,
            'r_squared': r_value ** 2
        }


def analyze_hedge_funds(HF_returns, factors):
    results = []
    for hf_name in HF_returns.keys()[1:]:  # Start from index 1 to skip the 'Date' column
        HF = hedgeFundClass(HF_returns[hf_name], hf_name)
        results.append(HF.run_capm_regression(factors))
    return pd.DataFrame(results)

time_Length = len(HF_returns)
hedge_fund_list = analyze_hedge_funds(HF_returns, Emp_factors['Mkt-RF'].iloc[-time_Length:])
display(hedge_fund_list)

# Emp_factors['Mkt-RF'].iloc[-time_Length:]
# print(HF_returns[HF_returns.keys()[i]])
# print(Emp_factors['Date'].iloc[-time_Length:])



HF_name      beta     alpha  \
0  HFRI 400 (US) Fund Weighted Composite Index (H...  0.002999  0.002638   
1      HFRI 400 (US) EH: Long/Short Index (HFRI4ELS)  0.005018  0.002149   
2  HFRI 400 (US) EH: Equity Market Neutral Index ...  0.000863  0.002246   
3  HFRI 400 (US) EH: Fundamental Value Index (HFR...  0.005566  0.001473   
4         HFRI 400 (US) Event-Driven Index (HFRI4ED)  0.003527  0.002325   
5                 HFRI 400 (US) Macro Index (HFRI4M)  0.000528  0.003226   
6       HFRI 400 (US) Relative Value Index (HFRI4RV)  0.001826  0.003223   

      t_stat  r_squared  
0  17.790518   0.651397  
1  10.875956   0.746495  
2  18.968367   0.195413  
3   8.108550   0.810910  
4  11.685065   0.589218  
5  14.584228   0.025349  
6  22.662552   0.429358

Based on the provided data, let's analyze and select two hedge fund strategies that could be considered attractive investment opportunities. The key metrics to analyze are beta, alpha, t-stat (significance of alpha), and r-squared (goodness of fit).

HFRI 400 (US) EH: Equity Market Neutral Index (HFRI4EMN)

Beta: 0.000863, indicating very low correlation with the market,    making it less sensitive to market movements.
Alpha: 0.002246, which reflects positive returns independent of the market.
t-stat: 18.968367, suggesting that the alpha is statistically significant.
R-squared: 0.195413, indicating that a small portion of returns is explained by market movements, which is expected for market-neutral strategies.
Why It’s Attractive:

The low beta makes this strategy appealing for diversification, as it has limited exposure to market risk.
The statistically significant alpha suggests a strong capability of generating excess returns independent of market movements.
It is ideal for investors looking to reduce market risk while aiming for stable returns.
HFRI 400 (US) Relative Value Index (HFRI4RV)

Beta: 0.001826, also indicating low market exposure.
Alpha: 0.003223, showing higher excess returns compared to other strategies in this dataset.
t-stat: 22.662552, indicating the alpha is highly significant.
R-squared: 0.429358, meaning a moderate portion of its returns is explained by market factors, but with significant additional sources of return.
Why It’s Attractive:

The combination of a low beta and the highest alpha makes this an attractive choice for investors seeking to generate strong risk-adjusted returns.
The high t-stat indicates that the positive alpha is not a result of randomness but a consistent ability to generate excess returns.
It is suitable for investors looking for income with some degree of market correlation but with a strong likelihood of outperforming the market.
These two strategies, the Equity Market Neutral Index for its diversification benefits and the Relative Value Index for its high alpha and significant returns, offer different but complementary benefits for an investment portfolio.

#Four-Factor Model